### ISSUE TITLE
Test start  BTP with  ICE chain

### ISSUE THREAD

**Issue by andrii-kl** [icon-project/icon-bridge#501]
## Overview

Try to start BTP with Ice chain (Arctic Testnet), using existing bsh functionality. 

## Story

As a developer, I want to be able deploy the btp contracts in to Arctic Testnet

## Acceptance Criteria

- BTP contracts deployed in to the Arctic Testnet

**Comment by andrii-kl** [comment]
```
1_deploy_bts.js
===============
Start deploy Proxy BTSCore 

   Deploying 'TransparentUpgradeableProxy'
   ---------------------------------------
   > transaction hash:    0x9b453333523316bcc6b0454008464841303763ef9868feb003670ca155cf8fdf
   > Blocks: 1            Seconds: 20
   > contract address:    0x6156f03388DAe58738d17024d874694A9afa30Fb
   > block number:        388233
   > block timestamp:     1663747536
   > account:             0x5AeAd9AF34c267aEb636092FE85dd710A6c8ab90
   > balance:             999.64468944
   > gas used:            808118 (0xc54b6)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.01616236 ETH

Start deploy Proxy BTSPeriphery 

Exiting: Review successful transactions manually by checking the transaction hashes above on Etherscan.


Error: invalid address (argument="address", value="", code=INVALID_ARGUMENT, version=address/5.6.1) (argument="_bmc", value="", code=INVALID_ARGUMENT, version=abi/5.6.4)
    at Logger.makeError (/usr/local/Cellar/truffle/5.5.20/libexec/lib/node_modules/truffle/build/webpack:/node_modules/@ethersproject/logger/lib.esm/index.js:219:1)

```

**Comment by andrii-kl** [comment]
```
  Deploying 'TransparentUpgradeableProxy'
   ---------------------------------------
   > transaction hash:    0xcadc89edab7429064453285371d3b1e83eeb972407d7df87c454de2c609dca29
   ⠼ Blocks: 1            Seconds: 23
/Users/anymac/projects/web3j/icon-bridge/devnet/docker/icon-bsc/build/contracts/solidity/bmc/node_modules/eth-block-tracker/src/polling.js:51
        const newErr = new Error(`PollingBlockTracker - encountered an error while attempting to update latest block:\n${err.stack}`)
                       ^
Error: PollingBlockTracker - encountered an error while attempting to update latest block:
undefined
    at PollingBlockTracker._performSync (/Users/anymac/projects/web3j/icon-bridge/devnet/docker/icon-bsc/build/contracts/solidity/bmc/node_modules/eth-block-tracker/src/polling.js:51:24)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by andrii-kl** [comment]
All contracts has been successfully deployed on the Arctic testnet (ICE) with small configuration BSC changes. 
Right now I'm working on fixing Relay block pooling issue.

**Comment by bbist** [comment]
Hi @andrii-kl,

After you deploy the contracts on both chains, you need to link the BMCs by first setting link to ICE's BMC in ICON's BMC, and then to ICON's BMC in ICE's BMC. And follow it by setting expected rxHeight on respective chains.

https://github.com/icon-project/icon-bridge/blob/f7ab8b419738a01854713c3e9f70492cda4ef778/.github/workflows/link-register-coin.yml#L126
https://github.com/icon-project/icon-bridge/blob/f7ab8b419738a01854713c3e9f70492cda4ef778/.github/workflows/link-register-coin.yml#L235

**Comment by bbist** [comment]
> ```
> 1_deploy_bts.js
> ===============
> Start deploy Proxy BTSCore 
> 
>    Deploying 'TransparentUpgradeableProxy'
>    ---------------------------------------
>    > transaction hash:    0x9b453333523316bcc6b0454008464841303763ef9868feb003670ca155cf8fdf
>    > Blocks: 1            Seconds: 20
>    > contract address:    0x6156f03388DAe58738d17024d874694A9afa30Fb
>    > block number:        388233
>    > block timestamp:     1663747536
>    > account:             0x5AeAd9AF34c267aEb636092FE85dd710A6c8ab90
>    > balance:             999.64468944
>    > gas used:            808118 (0xc54b6)
>    > gas price:           20 gwei
>    > value sent:          0 ETH
>    > total cost:          0.01616236 ETH
> 
> Start deploy Proxy BTSPeriphery 
> 
> Exiting: Review successful transactions manually by checking the transaction hashes above on Etherscan.
> 
> 
> Error: invalid address (argument="address", value="", code=INVALID_ARGUMENT, version=address/5.6.1) (argument="_bmc", value="", code=INVALID_ARGUMENT, version=abi/5.6.4)
>     at Logger.makeError (/usr/local/Cellar/truffle/5.5.20/libexec/lib/node_modules/truffle/build/webpack:/node_modules/@ethersproject/logger/lib.esm/index.js:219:1)
> ```

It'd be nice to have comments on how these issues got resolved, for posterity.

**Comment by andrii-kl** [comment]
- **Go-ethereum** library what we use to communicate with BSM and ICE, doesn't store a block hash , it calculate it in runtime. The problem is that BSM/Eth and ICE has different hash calculation algorithms, so if we recalculate block hash it will be different.
[https://github.com/ethereum/go-ethereum/blob/f651d6d56b4e7cc706f43074baf598813eb28644/core/types/block.go#L111](https://github.com/ethereum/go-ethereum/blob/f651d6d56b4e7cc706f43074baf598813eb28644/core/types/block.go#L111)

[https://github.com/paritytech/frontier/tree/master/frame/evm#evm-module-vs-ethereum-network](https://github.com/paritytech/frontier/tree/master/frame/evm#evm-module-vs-ethereum-network)

**Comment by CyrusVorwald-ICON** [comment]
@andrii-kl can you raise a new issue for this hash calculation problem?

**Comment by andrii-kl** [comment]
@CyrusVorwald-ICON  
https://app.zenhub.com/workspaces/icon-bridge-625892ed86f6a3001152d8f8/issues/icon-project/icon-bridge/556

**Comment by manishbista28** [comment]
Relay's BSC Chain API could not be used as it is and needed slight modification to a method.
Following is the git issue mentioning the problem.
https://github.com/icon-project/icon-bridge/issues/601
